docs/new-architecture-app-renderer-ios.md

```diff
@@ -0,0 +1,120 @@
+fabric_enabled = ENV['USE_FABRIC']
+use_codegen_discovery= ENV['USE_CODEGEN_DISCOVERY']
+
+# Enabled codegen discovery. This will run the codegen at preinstall time.
+# Files are generated at {pod installation root}/build/generated/ios/
+if use_codegen_discovery
+  Pod::UI.puts "[Codegen] Building target with codegen library discovery enabled."
+  pre_install do |installer|
+    use_react_native_codegen_discovery!({
+        react_native_path: config[:reactNativePath],
+        # Modify here if your app root path isn't the same as this one.
+        app_path: "#{Dir.pwd}/..",
+        fabric_enabled: fabric_enabled,
+    })
+  end
+end
+
+# Pass the flag to enable fabric to use_react_native!.
+use_react_native!(
+  ...
+  :fabric_enabled => fabric_enabled
+)
+```
+
+```bash
+$ ~/yourproject/**/ios; pod install
+```
+
+## 2. Update your root view
+
+The way to render your app with Fabric depends on your setup. Here is an example of how you can enable Fabric in your app with the RN_FABRIC_ENABLED compiler flag to enable/disable. Refer[RN-Tester’s AppDelegate](https://github.com/facebook/react-native/blob/main/packages/rn-tester/RNTester/AppDelegate.mm) as an example.
+
+```objc title="AppDelegate.mm"
+#ifdef RN_FABRIC_ENABLED
+#import <React/RCTFabricSurfaceHostingProxyRootView.h>
+#import <React/RCTSurfacePresenter.h>
+#import <React/RCTSurfacePresenterBridgeAdapter.h>
+#import <react/config/ReactNativeConfig.h>
+#endif
+
+@interface AppDelegate () <RCTCxxBridgeDelegate,
+                           RCTTurboModuleManagerDelegate> {
+#ifdef RN_FABRIC_ENABLED
+  RCTSurfacePresenterBridgeAdapter *_bridgeAdapter;
+  std::shared_ptr<const facebook::react::ReactNativeConfig> _reactNativeConfig;
+  facebook::react::ContextContainer::Shared _contextContainer;
+#endif
+
+  // Find a line that define rootView and replace/edit with the following lines.
+
+#ifdef RN_FABRIC_ENABLED
+  _contextContainer = std::make_shared<facebook::react::ContextContainer const>();
+  _reactNativeConfig = std::make_shared<facebook::react::EmptyReactNativeConfig const>();
+
+  _contextContainer->insert("ReactNativeConfig", _reactNativeConfig);
+
+  _bridgeAdapter = [[RCTSurfacePresenterBridgeAdapter alloc]
+        initWithBridge:_bridge
+      contextContainer:_contextContainer];
+
+  _bridge.surfacePresenter = _bridgeAdapter.surfacePresenter;
+
+  UIView *rootView =
+      [[RCTFabricSurfaceHostingProxyRootView alloc] initWithBridge:_bridge
+                                                        moduleName:@"MyTestApp"
+                                                 initialProperties:nil];
+#else
+  // Current implementation to define rootview.
+  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
+                                                   moduleName:@"MyTestApp"
+                                            initialProperties:nil];
+#endif
+```
+
+## 3. Add Babel Plugins
+
+This will trigger the codegen that will run at the metro building time.
```

Comment:
Can we make this part of the preset? E.g.
```javascript
module.exports = {
  presets: [
    ['module:metro-react-native-babel-preset', { codegen: true }],
  ],
};
```

Reply:
it would be great if it implements in this way.